src/docs.rs

```diff
@@ -153,7 +171,81 @@ pub fn generate_html(
     files
 }
 
-fn function(statement: &TypedStatement) -> Option<Function<'_>> {
+fn build_source_link_generator(
+    project_root: impl AsRef<Path>,
+    project_config: &PackageConfig,
+    module_path: &PathBuf,
+    name: &String,
+) -> Box<dyn SourceLinkGenerator> {
+    match (&project_config.version, &project_config.repository) {
```

Comment:
We have a problem here in that there's nothing that guarentees that the version in `gleam.toml` matches the version in `.app`. This will be very error prone.

Perhaps we should make `version` a required field and then add some code that checks `.app` to see if it is the same version as `gleam.toml`. We don't have a parser for `.app` yet so can just check it's a substring to start.

Reply:
Can certainly try to do that. It also means that the app version, gleam version and git tag should probably all match. `gleam/http` uses a `v` prefix for the tag but not the app file version.

Reply:
I've attempted this. Happy to have feedback. I guess it isn't great as I'm using the `?` operator and returning a result to early exit from the calling function. Probably could be better.